command/kv_patch.go

```diff
@@ -147,8 +159,8 @@ func (c *KVPatchCommand) Run(args []string) int {
 	case len(args) < 1:
 		c.UI.Error(fmt.Sprintf("Not enough arguments (expected >1, got %d)", len(args)))
 		return 1
-	case len(args) == 1:
-		c.UI.Error("Must supply data")
+	case len(c.flagRemoveData) == 0 && len(args) == 1:
```

Comment:
Oh, great change to this check! I can definitely imagine scenarios where users only want to remove data.